pep-0695.rst

```diff
@@ -365,13 +377,16 @@ At runtime, a ``type`` statement will generate an instance of
 include:
 
 * ``__name__`` is a str representing the name of the type alias
-* ``__parameters__`` is a tuple of ``TypeVar``, ``TypeVarTuple``, or
+* ``__type_params__`` is a tuple of ``TypeVar``, ``TypeVarTuple``, or
   ``ParamSpec`` objects that parameterize the type alias if it is generic
 * ``__value__`` is the evaluated value of the type alias
 
-The ``__value__`` attribute initially has a value of ``None`` while the type
-alias expression is evaluated. It is then updated after a successful evaluation.
-This allows for self-referential type aliases.
+All of these attributes are read-only.
+
+The value of the type alias is evaluated lazily. Access to the ``__value__`` attribute
+triggers evaluation of the value and caches the resulting value.
+This allows for self-referential and mutually recursive type aliases, and it
+should largely eliminate the need for quoted forward references in type aliases.
```

Comment:
From in-person conversation there's an open question about whether `repr()` forces this evaluation, right?

Reply:
Do you think that should be specified in the PEP?

Reply:
But my takeaway from the conversation was that repr() should be something like this:
```
def __repr__(self):
    try:
        value = repr(self.__value__)
    except Exception:
        value = "an exception occurred while evaluating the value"
    return f"<type alias {self.__name__}: {value}>"
```